posthog/models/dashboard.py

```diff
@@ -1,15 +1,24 @@
-from typing import Any, Dict
+from ctypes import cast
+from typing import TYPE_CHECKING, Any, Dict
 
 from django.contrib.postgres.fields import ArrayField
 from django.db import models
 
+from posthog.constants import AvailableFeature
+
+if TYPE_CHECKING:
+    from posthog.models.user import User
+
 
 class Dashboard(models.Model):
-    CREATION_MODE_CHOICES = (
-        ("default", "Default"),
-        ("template", "Template"),  # dashboard was created from a predefined template
-        ("duplicate", "Duplicate"),  # dashboard was duplicated from another dashboard
-    )
+    class CreationMode(models.TextChoices):
+        DEFAULT = "default", "Default"
+        TEMPLATE = "template", "Template"  # dashboard was created from a predefined template
+        DUPLICATE = "duplicate", "Duplicate"  # dashboard was duplicated from another dashboard
+
+    class RestrictionLevel(models.IntegerChoices):
+        INHERENT_VIEW_AND_EDIT = 21, "Everyone in the project can edit"
```

Comment:
We've got three definitions of the choice numbers

```
(21, "Everyone in the project can edit"),
(37, "Only those invited to this dashboard can edit"),

InherentViewAndEdit = 21,
InherentViewButExplicitEdit = 37,

INHERENT_VIEW_AND_EDIT = 21, "Everyone in the project can edit"
INHERENT_VIEW_BUT_EXPLICIT_EDIT = 37, "Only those invited to this dashboard can edit"
```

And 2 ways of describing them

I found 

```
InherentViewAndEdit = 21,
InherentViewButExplicitEdit = 37,
```

much harder to parse the meaning of than

```
(21, "Everyone in the project can edit"),
(37, "Only those invited to this dashboard can edit"),
```

I'd be tempted to replace

`InherentViewAndEdit` -> `EveryoneInProjectCanEdit`
`InherentViewButExplicitEdit` -> `OnlyInvitedUsersCanEdit` or `OnlySpecifiedUsersCanEdit` or similar

Reply:
That's a fair point. I went for the more technical naming because I wanted to reuse `RestrictionLevel` in `DashboardPrivilege`, so I wanted the wording to work both for the dashboard and for the user (avoiding the need to sync two enums), but I can see how readability suffers for this. I'll just rename these as you suggested and add an analogous `PrivilegeLevel` enum which will be equally readable, but for the user side.

Reply:
```
/** Collaboration restriction level (which is a dashboard setting). Sync with DashboardPrivilegeLevel. */
export enum DashboardRestrictionLevel {
    EveryoneInProjectCanEdit = 21,
    OnlyCollaboratorsCanEdit = 37,
}

/** Collaboration privilege level (which is a user property). Sync with DashboardRestrictionLevel. */
export enum DashboardPrivilegeLevel {
    CanView = 21,
    CanEdit = 37,
}
```

👌